language/move-lang/src/typing/core.rs

```diff
@@ -732,15 +732,42 @@ pub fn make_function_type(
     };
     let defined_loc = finfo.defined_loc;
     match &finfo.visibility {
-        FunctionVisibility::Internal if !in_current_module => {
-            let internal_msg = "This function is internal to its module. Only 'public' functions \
+        FunctionVisibility::Internal => {
+            if !in_current_module {
+                let internal_msg =
+                    "This function is internal to its module. Only 'public' functions \
                                 can be called outside of their module";
-            context.error(vec![
-                (loc, format!("Invalid call to '{}::{}'", m, f)),
-                (defined_loc, internal_msg.into()),
-            ])
+                context.error(vec![
+                    (loc, format!("Invalid call to '{}::{}'", m, f)),
+                    (defined_loc, internal_msg.into()),
+                ])
+            }
         }
-        _ => (),
+        FunctionVisibility::Script(_) => {
+            let in_script_context = match &context.current_module {
+                Some(current_m) => {
+                    let current_f = context
+                        .current_function
+                        .as_ref()
+                        .expect("ICE must be in a function to call other functions");
```

Comment:
You can't assume this. Someone could have called a function in a constant and this will cause a panic

Reply:
Thanks for pointing out this! I forgot the constant case. I'll fix it.